### PR TITLE
fix: adds refresh logic when one connection is expired

### PR DESCRIPTION
**Pull request by jaypalmudaliyar24**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: lib/mobility-core/src/Kernel/Storage/Esqueleto/Transactionable.hs [hunk 1]
```diff
@@ -104,4 +116,26 @@ runNoTransactionIO logEnv dbEnv (SqlDB run) = do
...
           { currentTime = now
           }
-  runLoggerIO logEnv $ runSqlPoolNoTransaction (runReaderT run sqlDBEnv) dbEnv.connPool Nothing
+  runLoggerIO logEnv $ do
+    res <- try @_ @SomeException $ runSqlPoolNoTransaction (runReaderT run sqlDBEnv) dbEnv.connPool Nothing
+    fromEitherOrM res $ \e -> do
+      let res' = transformException e
+      logTagError "DB QUERY" $ "Transformed ERROR response: " <> show res'
+      case res' of
+        ET.DBError (ET.SQLError (ET.PostgresError (ET.PostgresSqlError "" ET.PostgresFatalError "" "" ""))) _ -> do
```
Comment (roman-bodavskiy) [inline lib/mobility-core/src/Kernel/Storage/Esqueleto/Transactionable.hs:125]: this case will match only when all these fields will be empty
```
       sqlState = ""
    , sqlErrorMsg = ""
    , sqlErrorDetail = ""
    , sqlErrorHint  = ""
```
Is it really ever happens?
Maybe you meant this:
`ET.PostgresSqlError _ ET.PostgresFatalError _ _ _)`

Comment (jaypalmudaliyar24) [reply lib/mobility-core/src/Kernel/Storage/Esqueleto/Transactionable.hs:125]: No actually this is the exact case when db connection expires. 
```
beckn-rider-dashboard-master-2ad616-76fff5645b-vj4zf beckn-rider-dashboard-master {"log":"2024-12-17 11:02:50.314332966 UTC ERROR> @beckn-rider-dashboard-master-2ad616-76fff5645b-vj4zf [requestId-9a895842-4e09-416c-ba91-d63c9c689ceb, DB QUERY] |> Transformed ERROR response: DBError (SQLError (PostgresError (PostgresSqlError {sqlState = \"\", sqlExecStatus = PostgresFatalError, sqlErrorMsg = \"\", sqlErrorDetail = \"\", sqlErrorHint = \"\"}))) \"SqlError {sqlState = \\\"\\\", sqlExecStatus = FatalError, sqlErrorMsg = \\\"\\\", sqlErrorDetail = \\\"\\\", sqlErrorHint = \\\"\\\"}\""}
```

File: lib/mobility-core/src/Kernel/Storage/Esqueleto/Transactionable.hs [hunk 2]
```diff
@@ -104,4 +116,26 @@ runNoTransactionIO logEnv dbEnv (SqlDB run) = do
         SqlDBEnv
           { currentTime = now
           }
-  runLoggerIO logEnv $ runSqlPoolNoTransaction (runReaderT run sqlDBEnv) dbEnv.connPool Nothing
+  runLoggerIO logEnv $ do
+    res <- try @_ @SomeException $ runSqlPoolNoTransaction (runReaderT run sqlDBEnv) dbEnv.connPool Nothing
+    fromEitherOrM res $ \e -> do
```
Comment (roman-bodavskiy) [inline lib/mobility-core/src/Kernel/Storage/Esqueleto/Transactionable.hs:121]: can be simplified:

`    catch @_ @SomeException (runSqlPool (runReaderT run sqlDBEnv) dbEnv.connPool) $ \e -> do`
then `fromEitherOrM` not required

**Review by roman-bodavskiy — APPROVED** [review]

File: flake.lock [hunk 3]
```diff
@@ -370,11 +370,11 @@
...
       },
       "locked": {
-        "lastModified": 1734090572,
-        "narHash": "sha256-DwvTuZFjrStKYOxZn99s7Fk89AZQskQ0593AexqY7lc=",
+        "lastModified": 1734428345,
+        "narHash": "sha256-Nbss1aAIfrgNMinY0AOfaOa8EUXWhAxHrw7Fhyb55d0=",
         "owner": "nammayatri",
         "repo": "euler-hs",
-        "rev": "dfe2249ca40fbee35b772eaa937c762498309312",
+        "rev": "bccffdbc6c27e9ed648a1abb95f288763cff6bcc",
```
Comment (roman-bodavskiy) [inline flake.lock:377]: looks like wrong commit

Comment (jaypalmudaliyar24) [reply flake.lock:377]: we are pointing to this branch currently in main. 
https://github.com/nammayatri/euler-hs/commits/backend/logging-latency/

**Review by arjuna92 — APPROVED** [review]

**Pull request merged**